nepomuk/src/main/java/edu/asu/diging/gilesecosystem/nepomuk/core/service/impl/SystemMessageHandler.java

```diff
@@ -66,6 +70,7 @@ public void handleError(String msg, Exception exception) {
             requestProducer.sendRequest(request, propertiesManager.getProperty(Properties.KAFKA_TOPIC_SYSTEM_MESSAGES));
         } catch (MessageCreationException e) {
             logger.error("Could not send request.", e);
+            systemMessageHandler.handleError("Could not send request.", e);
         }
     }
 }
```

Comment:
this will create an infinite loop. If the system message handler can't put messages in the pipeline, it has to just log them. Othwerwise, it'll keep on calling itself over and over again.